src/main/java/updater/JarReplaceAndExecute.java

```diff
@@ -0,0 +1,193 @@
+package updater;
+
+import javafx.application.Application;
+import javafx.application.Platform;
+import javafx.geometry.Insets;
+import javafx.scene.Group;
+import javafx.scene.Scene;
+import javafx.scene.control.Label;
+import javafx.scene.layout.VBox;
+import javafx.stage.Stage;
+
+import java.io.*;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.StandardCopyOption;
+import java.util.HashMap;
+
+/**
+ * Note: This class will be compiled into a JAR on its own
+ *
+ * Replaces a JAR file and start that JAR
+ */
+public class JarReplaceAndExecute extends Application{
+    private static final String BACKUP_FILENAME_APPENDIX = "_backup";
+
+    @SuppressWarnings("PMD")
```

Comment:
What's the warning being suppressed? Should be in a comment since it's not always obvious to readers.